### DISCUSSION TITLE
Is there any way to inherit a class from SelectableTextBlock?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#17231] [Q&A] [answered]
The `SelectableTextBlock` class is not sealed and has protected methods. So presumably it is permissible to inherit from it?

However, this does not work...

`public class AvTextBlock2 : SelectableTextBlock`

The text remains unresponsive to selection.

Any way to resolve this?

**Comment by stevemonaco** [comment, marked as answer]
If you check DevTools, the selection is actually being made. Even though `SelectableTextBlock` is a primitive control that handles its own rendering, it still requires some extra help. The `SelectionBrush` is null by default unless you set it. So you can either write a new style, set it explicitly, or use `StyleKeyOverride` so your theme will do the work. eg.

```xml
public class AvTextBlock2 : SelectableTextBlock
{
    protected override Type StyleKeyOverride => typeof(SelectableTextBlock);
}
```

**Reply by kuiperzone** [reply]
Working now. Thank you!
